Refactor ReadableStream like done for WritableStream

youennf opened the pull request:
#### 647a5885560717c27e03544ee9a39ee3e8035d38
<pre>
Refactor ReadableStream like done for WritableStream
<a href="https://bugs.webkit.org/show_bug.cgi?id=248307">https://bugs.webkit.org/show_bug.cgi?id=248307</a>
rdar://problem/102648417

Reviewed by Alex Christensen.

This improves consistency between the two code paths and allows to reuse more of the binding generator.
A future step will be to add async iterable to ReadableStream.
This is a rebasing of <a href="https://commits.webkit.org/257174@main.">https://commits.webkit.org/257174@main.</a>

Covered by existing tests.

* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.worker-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::clone):
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
(WebCore::FetchBodyOwner::createReadableStream):
* Source/WebCore/Modules/fetch/FetchResponse.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::RTCRtpSFrameTransform::createStreams):
* Source/WebCore/Modules/streams/ReadableStream.cpp: Added.
(WebCore::ReadableStream::create):
(WebCore::ReadableStream::createFromJSValues):
(WebCore::ReadableStream::ReadableStream):
(WebCore::ReadableStream::tee):
(WebCore::JSReadableStream::cancel):
(WebCore::JSReadableStream::getReader):
(WebCore::JSReadableStream::pipeTo):
(WebCore::JSReadableStream::pipeThrough):
* Source/WebCore/Modules/streams/ReadableStream.h: Added.
(WebCore::ReadableStream::lock):
(WebCore::ReadableStream::isLocked const):
(WebCore::ReadableStream::isDisturbed const):
(WebCore::ReadableStream::cancel):
(WebCore::ReadableStream::pipeTo):
(WebCore::ReadableStream::internalReadableStream):
* Source/WebCore/Modules/streams/ReadableStream.idl:
* Source/WebCore/Modules/streams/ReadableStream.js: Removed.
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.js:
(initializeReadableStreamBYOBReader):
* Source/WebCore/Modules/streams/ReadableStreamDefaultReader.js:
(initializeReadableStreamDefaultReader):
* Source/WebCore/Modules/streams/ReadableStreamInternals.js:
(createInternalReadableStreamFromUnderlyingSource):
(readableStreamGetReaderForBindings):
(readableStreamCancelForBindings):
(readableStreamPipeThroughForBindings):
(readableStreamPipeToForBindings):
(readableStreamTee):
* Source/WebCore/Modules/streams/TransformStreamDefaultController.js:
(getter.desiredSize):
* Source/WebCore/Modules/streams/TransformStreamInternals.js:
(initializeTransformStream):
(transformStreamError):
(transformStreamDefaultControllerEnqueue):
(transformStreamDefaultControllerTerminate):
(transformStreamDefaultSinkCloseAlgorithm):
* Source/WebCore/Modules/streams/WritableStream.idl:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/InternalReadableStream.cpp: Added.
(WebCore::invokeReadableStreamFunction):
(WebCore::InternalReadableStream::createFromUnderlyingSource):
(WebCore::InternalReadableStream::fromObject):
(WebCore::InternalReadableStream::isLocked const):
(WebCore::InternalReadableStream::isDisturbed const):
(WebCore::InternalReadableStream::cancel):
(WebCore::InternalReadableStream::lock):
(WebCore::InternalReadableStream::pipeTo):
(WebCore::InternalReadableStream::tee):
(WebCore::InternalReadableStream::getReader):
(WebCore::InternalReadableStream::pipeThrough):
* Source/WebCore/bindings/js/InternalReadableStream.h: Added.
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMGlobalObject::addBuiltinGlobals):
* Source/WebCore/bindings/js/ReadableStream.cpp: Removed.
* Source/WebCore/bindings/js/ReadableStream.h: Removed.
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(AddToIncludesForIDLType):
* Source/WebCore/fileapi/Blob.cpp:
(WebCore::Blob::stream):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isReadableStreamDisturbed):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: <a href="https://commits.webkit.org/262590@main">https://commits.webkit.org/262590@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c99f39bb230c8db21cb89dbe9175002104e9b639

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1960 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1991 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2053 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2047 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1936 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2036 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1807 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1979 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1766 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1759 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2733 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1756 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1741 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1653 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1803 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1788 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2904 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1815 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1616 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1742 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1752 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/489 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1906 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/bindings/js/InternalReadableStream.cpp:
```diff
@@ -0,0 +1,282 @@
+    return result.hasException() ? false : result.returnValue().isTrue();
+}
+
+bool InternalReadableStream::disturbed() const
```
I think isDisturbed, isLocked, etc. are better names.

youennf committed:
Refactor ReadableStream like done for WritableStream
https://bugs.webkit.org/show_bug.cgi?id=248307
rdar://problem/102648417

Reviewed by Alex Christensen.

This improves consistency between the two code paths and allows to reuse more of the binding generator.
A future step will be to add async iterable to ReadableStream.
This is a rebasing of https://commits.webkit.org/257174@main.

Covered by existing tests.

* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/streams/idlharness.any.worker-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::clone):
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
(WebCore::FetchBodyOwner::createReadableStream):
* Source/WebCore/Modules/fetch/FetchResponse.cpp:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::RTCRtpSFrameTransform::createStreams):
* Source/WebCore/Modules/streams/ReadableStream.cpp: Added.
(WebCore::ReadableStream::create):
(WebCore::ReadableStream::createFromJSValues):
(WebCore::ReadableStream::ReadableStream):
(WebCore::ReadableStream::tee):
(WebCore::JSReadableStream::cancel):
(WebCore::JSReadableStream::getReader):
(WebCore::JSReadableStream::pipeTo):
(WebCore::JSReadableStream::pipeThrough):
* Source/WebCore/Modules/streams/ReadableStream.h: Added.
(WebCore::ReadableStream::lock):
(WebCore::ReadableStream::isLocked const):
(WebCore::ReadableStream::isDisturbed const):
(WebCore::ReadableStream::cancel):
(WebCore::ReadableStream::pipeTo):
(WebCore::ReadableStream::internalReadableStream):
* Source/WebCore/Modules/streams/ReadableStream.idl:
* Source/WebCore/Modules/streams/ReadableStream.js: Removed.
* Source/WebCore/Modules/streams/ReadableStreamBYOBReader.js:
(initializeReadableStreamBYOBReader):
* Source/WebCore/Modules/streams/ReadableStreamDefaultReader.js:
(initializeReadableStreamDefaultReader):
* Source/WebCore/Modules/streams/ReadableStreamInternals.js:
(createInternalReadableStreamFromUnderlyingSource):
(readableStreamGetReaderForBindings):
(readableStreamCancelForBindings):
(readableStreamPipeThroughForBindings):
(readableStreamPipeToForBindings):
(readableStreamTee):
* Source/WebCore/Modules/streams/TransformStreamDefaultController.js:
(getter.desiredSize):
* Source/WebCore/Modules/streams/TransformStreamInternals.js:
(initializeTransformStream):
(transformStreamError):
(transformStreamDefaultControllerEnqueue):
(transformStreamDefaultControllerTerminate):
(transformStreamDefaultSinkCloseAlgorithm):
* Source/WebCore/Modules/streams/WritableStream.idl:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/InternalReadableStream.cpp: Added.
(WebCore::invokeReadableStreamFunction):
(WebCore::InternalReadableStream::createFromUnderlyingSource):
(WebCore::InternalReadableStream::fromObject):
(WebCore::InternalReadableStream::isLocked const):
(WebCore::InternalReadableStream::isDisturbed const):
(WebCore::InternalReadableStream::cancel):
(WebCore::InternalReadableStream::lock):
(WebCore::InternalReadableStream::pipeTo):
(WebCore::InternalReadableStream::tee):
(WebCore::InternalReadableStream::getReader):
(WebCore::InternalReadableStream::pipeThrough):
* Source/WebCore/bindings/js/InternalReadableStream.h: Added.
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMGlobalObject::addBuiltinGlobals):
* Source/WebCore/bindings/js/ReadableStream.cpp: Removed.
* Source/WebCore/bindings/js/ReadableStream.h: Removed.
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(AddToIncludesForIDLType):
* Source/WebCore/fileapi/Blob.cpp:
(WebCore::Blob::stream):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isReadableStreamDisturbed):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/262590@main
